### DISCUSSION TITLE
Add option for spread rest siblings exclusion for `noUnusedVariables` rule

### DISCUSSION THREAD

**Discussion by EvHaus** [biomejs/biome#572] [Ideas]
A common pattern to omit keys from an object is to spread it like so:

```js
const car = {model: "Model S", brand: "Tesla", year: 2019};

// Remove model & year keys from the object
const {model, year, ...other} = car;

// Use the new object
formatBrand(other);
```

This is effectively equivalent to:

```js
const car = {model: "Model S", brand: "Tesla", year: 2019};

delete car.model;
delete car.year;

// Use the new object
formatBrand(car);
```

Except that it's an immutable way of doing it, and doesn't violate the `performance.noDelete` rule.

Unfortunately, the `correctness/noUnusedVariables` rule currently (and correctly) flags this syntax as a violation, as the `model` and `year` variables now become unused.

I'm wondering if maybe there could be an option on the `correctness/noUnusedVariables` rule to `ignoreRestSiblings` or something like that? ESLint has it here: https://eslint.org/docs/latest/rules/no-unused-vars#ignorerestsiblings

**Comment by Conaclos** [comment]
Thanks for bringing this to light! I think this could be the default behavior without any option. We already skip some cases like unused parameters preceding a used parameter. It's kind of related.

**Comment by chungweileong94** [comment]
Yeah, this one feels like a deal breaker to me. The only way to work around this is to label it with `_`.

**Comment by Conaclos** [comment]
I opened #1910 to address this.
